### DISCUSSION TITLE
And the correct SQL Server database.url is... ?

### DISCUSSION THREAD

**Discussion by Coooz** [nextgenhealthcare/connect#5475] [Q&A]
Hi,

My Mirth Connect database is called _MirthOruProcessing_, which resides on a database server called _zrg06\ins03_ .
Having specificied _com.microsoft.sqlserver.jdbc.SQLServerDriver_ as database.driver in mirth.properties, I only need to come up with the right database.url in order to get the connection working.
I got it working once, but I needed to reinstall Mirth Connect (3.12) and foolishly forgot to backup mirth.properties.

These are incorrect:
#database.url = jdbc:sqlserver://zrg06//ins03;databaseName=MirthOruProcessing
#database.url = jdbc:sqlserver://zrg06/ins03;databaseName=MirthOruProcessing
#database.url = jdbc:sqlserver://zrg06//ins03:1433;databaseName=MirthOruProcessing
#database.url = jdbc:sqlserver://zrg06/ins03:1433;databaseName=MirthOruProcessing
#database.url = jdbc:sqlserver://zrg06\\ins03:1433;databaseName=MirthOruProcessing
#database.url = jdbc:sqlserver://zrg06\\ins03;databaseName=MirthOruProcessing

Anyone any suggestions as to what the correct one might be?
Thx!

**Comment by jonbartels** [comment]
Is your server literally called `zrg06\ins03` or is the server `zrg06` and the instance is `ins03`?

https://learn.microsoft.com/en-us/sql/connect/jdbc/building-the-connection-url?view=sql-server-ver16

If the server is literally called `zrg06\ins03` you might need to escape the backslash: `database.url = jdbc:sqlserver://zrg06\\ins03:1433;databaseName=MirthOruProcessing`. 

If its `server\instance` then: `database.url = jdbc:sqlserver://zrg06\ins03:1433;databaseName=MirthOruProcessing`. 

A trick I like to use to figure out JDBC connections is to take a running MC instance, even one on Derby, and connect to the database using a DB Reader. That lets me iterate quickly, without DB restarts, and work out the connection string details.

**Reply by Coooz** [reply]
Hi @jonbartels,
I was able to get it working with the jtds variation:
`jdbc:jtds:sqlserver://zrg06:1433/MirthOruProcessing;instance=ins03`
I hesitated to use jtds since it is outdated, but since it is working now I'll just take that for granted I guess...

"If its `server\instance` then: `database.url = jdbc:sqlserver://zrg06\ins03:1433;databaseName=MirthOruProcessing`."
I tried that one, but it didn't seem to work either. Also I noticed that filling in this url in MC server manager made it appear in mirth.properties with the backslash escaped anyway.

**Comment by Otisssss** [comment]
Installing actual driver just did it and worked. Following these steps

MS SQL Actual ODBC Driver
Get driver, install to mirth database folder 
Visit here to get trusted certs from MS: https://www.microsoft.com/pki/mscorp/cps/default.htm
Get the two trusted certs to ensure that the ssl works
Import these two certs to jvm trust store. Use portecle or run keystore command. Must be run as admin 
Add the driver and connect settings in one of the database reader/writers to save the xml config
Sample string to connect..
jdbc:sqlserver://DB:port;databaseName=DBToConnect;encrypt=true;trustServerCertificate=true

**Reply by pacmano1** [reply]
See https://github.com/pacmano1/DiridiumTechnologies/blob/main/mssql_jdbc_connect_strings, corrections welcome.

**Reply by tonygermano** [reply]
I think missing from your answer is how to specify the instance name. Also starting with mirth 3.10, the MS driver is included and you do not need to install it.

**Reply by Coooz** [reply]
@Otisssss Thanks for your elaborate answer, but I actually do not want to do that, since I got it working once by specifying the correct url only. I'm very hesitant to install and configure things when that should not be necessary.

**Reply by Coooz** [reply]
@pacmano1
Thanks for all of your suggestions! Unfortunately I do not have access to the environment in which the issue occurred anymore. I got it working using the jtds variation and it looks like others took it from there without me now.
Once I will be able to return there, I can try them out and see what works. I'll keep that in mind.

**Reply by Coooz** [reply]
@tonygermano 
Thank you. Specifying the instance incorrectly indeed was probably it.

**Reply by Coooz** [reply]
@pacmano1 
I was able to have a look at it once more this morning. I checked out the page in your link. Since I have to use SQL Server authentication, I cannot say anything about the other authentication formats. 
But I tried both
`jdbc:sqlserver://zrg06:1433;databaseName=MososMirth_OruProcessing;instanceName=ins03`
which as such is actually not on the page, and
`jdbc:sqlserver://zrg06:1433;databaseName=MososMirth_OruProcessing;instanceName=ins03;user=<USER>;password=<PASSWORD>`
Both did not work. 

I doubt as well whether the `user=<USER>;password=<PASSWORD>` parameters need to be specified in the url since they are mentioned elsewhere in mirth.properties. Also the jtds variation that is working does not contain user and password parameters.

**Reply by pacmano1** [reply]
I posted generic jdbc connection strings, not mirth specific.
